### PR TITLE
Add command line migration

### PR DESCRIPTION
**Pull request by Javier-Rotelli**
since most of the flarum installations have access to the console, added an option to run the migration from it.

It may need some revisions to the options and its default values, but right now it's functional.

### CODE REVIEW

**Comment by MichaelBelgium** [comment]
Could you add some information/explanation about the command in the readme?

**Comment by Javier-Rotelli** [comment]
hey, sure. I didn't add it because I wanted to be sure yopu were alright with this approach

**Comment by MichaelBelgium** [comment]
I didn't test it yet but having a command is a nice addition, although i'm not sure if it's easier with all these options than in the browser.

**Comment by Javier-Rotelli** [comment]
Added the command line to the Readme. let me know if it isn't enough. 
yeah, it's not easier at first, but if anything happens (and in my experience something always happens 😬 ) the feedback is easier to follow, and once you wrote the command you don't need to write it again.

File: src/Commands/MybbToFlarumCommand.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+use Michaelbelgium\Mybbtoflarum\Migrator;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Question\Question;
+
+class MybbToFlarumCommand extends AbstractCommand
+{
+    protected $options = [
+        'host'=> ['host', null, InputOption::VALUE_REQUIRED, 'host of the mybb database'],
+        'user'=> ['user', 'u', InputOption::VALUE_REQUIRED, 'user of the mybb database'],
+        'password'=> ['password', 'p', InputOption::VALUE_OPTIONAL, 'password for the mybb database', false],
```
Comment (MichaelBelgium) [inline src/Commands/MybbToFlarumCommand.php:16]: Is `false` the default value? Would be better if it's an empty string

File: src/Commands/MybbToFlarumCommand.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        try {
+            $migrator = new Migrator(
+                $this->input->getOption('host'),
+                $this->input->getOption('user'),
+                $password,
+                $this->input->getOption('db'),
+                $this->input->getOption('prefix'),
+                $this->input->getOption('path')
+            );
```
Comment (MichaelBelgium) [inline src/Commands/MybbToFlarumCommand.php:65]: Can you make sure that the options "host", "user" and "db" are always required and no empty strings? There is InputOption::VALUE_REQUIRED but if you do the interactive mode it's not checking if it's filled in somehow ...

File: src/Commands/MybbToFlarumCommand.php [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        }
+        return $value;
+    }
+
+    protected function showCounts($migrator)
+    {
+        $counts = $migrator->getProcessedCount();
+        $this->info("• {$counts["users"]} users\n• {$counts["groups"]} user groups\n• {$counts["categories"]} categories\n• {$counts["discussions"]} discussions\n• {$counts["posts"]} posts");
+    }
+}
```
Comment (MichaelBelgium) [inline src/Commands/MybbToFlarumCommand.php:120]: Not really a fan of calling this function every time after it migrated something.

Could u change it so it'd look like, for example:

```
$this->info("2 user groups migrated");
$this->info("15 discussions migrated")
...;
```

Also with the use of `$migrator->getProcessedCount()` ofcourse

**Review by MichaelBelgium — CHANGES_REQUESTED** [review]
Sorry for the late check/review but if these changes are resolved you'll get a merge 😛

**Comment by Javier-Rotelli** [comment]
hey! better late than never!. I'll address your comments during the weekend

File: src/Commands/MybbToFlarumCommand.php (hunk 1 above)
Comment (Javier-Rotelli) [reply src/Commands/MybbToFlarumCommand.php]: done

File: src/Commands/MybbToFlarumCommand.php (hunk 2 above)
Comment (Javier-Rotelli) [reply src/Commands/MybbToFlarumCommand.php]: done

File: src/Commands/MybbToFlarumCommand.php (hunk 3 above)
Comment (Javier-Rotelli) [reply src/Commands/MybbToFlarumCommand.php]: I'm not 100% sure I got you here. but take a look at the new version and let me know what you think

**Comment by Javier-Rotelli** [comment]
well, took me long enough to get to the weekend 😬  hope this is ok.

File: src/Commands/MybbToFlarumCommand.php (hunk 3 above)
Comment (MichaelBelgium) [reply src/Commands/MybbToFlarumCommand.php]: Yeah almost, I only wanted to have "x user groups migrated" and not 

> user groups migrated

> x user groups migrated

one time is fine with the count

**Review by MichaelBelgium — APPROVED** [review]

**Pull request merged**